### PR TITLE
feat(text editor): block pasted inline images

### PR DESCRIPTION
**Pull request by john-traas**
Updates the link plugin which handles pasting to purge images from content pasted into the text editor. 

Fixes https://github.com/Lundalogik/crm-feature/issues/4152

### CODE REVIEW

File: src/components/text-editor/prosemirror-adapter/plugins/link-plugin.ts
Comment (civing) [inline src/components/text-editor/prosemirror-adapter/plugins/link-plugin.ts:1]: I'm thinking that we maybe should create a new plugin file for images? 🤔 
As long as we register that one before the link one it should be fine. What do you think?

File: src/components/text-editor/prosemirror-adapter/plugins/link-plugin.ts [hunk 1]
```diff
@@ -177,9 +177,66 @@ const processDoubleClickEvent = (
...
+        if (node.type.name === 'image') {
+            return true;
+        }
+
+        let found = false;
+        node.content.forEach((child) => {
+            if (isImageNode(child)) {
+                found = true;
+            }
+        });
```
[the remark this answers is not in the document]
Comment (john-traas) [reply src/components/text-editor/prosemirror-adapter/plugins/link-plugin.ts:196]: I think I was in a hurry and defaulted to using the variable. We could just return true within the forEach for the children and return false at the end instead.

File: src/components/text-editor/prosemirror-adapter/plugins/link-plugin.ts
Comment (civing) [reply src/components/text-editor/prosemirror-adapter/plugins/link-plugin.ts:1]: What do you think about this @john-traas ? https://github.com/Lundalogik/lime-elements/pull/3005/commits/c650433b75e9a2831c6aa885c2426c04c85f3e2f

**Review by civing — APPROVED** [review]

**Review by civing — APPROVED** [review]

File: src/components/text-editor/prosemirror-adapter/plugins/link-plugin.ts
Comment (john-traas) [reply src/components/text-editor/prosemirror-adapter/plugins/link-plugin.ts:1]: It looks really good. Much cleaner as a separate plugin!

**Pull request merged**